### PR TITLE
More attribute coverage and improved tests

### PR DESCRIPTION
**Pull request by bhelx**
* Added more coverage of fields
* Improved tests
* Changed Money currencies to lowercase symbols

### CODE REVIEW

File: lib/recurly/coupon.ex [hunk 1]
```diff
@@ -14,6 +14,8 @@ defmodule Recurly.Coupon do
     field :applies_to_non_plan_charges, :boolean
     field :coupon_code,                 :string
     field :coupon_type,                 :string
+    field :created_at,                  :date_time
+    field :deleted_at,                  :date_time
```
Comment (csmb) [inline lib/recurly/coupon.ex]: I think `created_at`, `updated_at`, and `deleted_at` should each include `read_only: true`.

Comment (bhelx) [reply lib/recurly/coupon.ex]: true

**Comment by csmb** [comment]
Looks great. What was your motivation for changing Money currencies to lowercase symbols?

**Comment by bhelx** [comment]
Struct fields must be lowercase:

```
iex(1)> m = %Recurly.Money{USD: 100}
%Recurly.Money{AUD: nil, BRL: nil, CAD: nil, CHF: nil, CZK: nil, DKK: nil,
 EUR: nil, GBP: nil, HUF: nil, ILS: nil, INR: nil, MXN: nil, NOK: nil, NZD: nil,
 PLN: nil, SEK: nil, SGD: nil, USD: 100, ZAR: nil, __meta__: nil}
iex(2)> Map.get(m, :USD)
100
iex(3)> m.USD
** (CompileError) iex:2: invalid alias: "m.USD". If you wanted to define an alias, an alias must expand to an atom at compile time but it did not, you may use Module.concat/2 to build it at runtime. If instead you wanted to invoke a function or access a field, wrap the function or field name in double quotes
```

With this branch:

```
iex(1)> m = %Recurly.Money{usd: 100}
%Recurly.Money{__meta__: nil, aud: nil, brl: nil, cad: nil, chf: nil, czk: nil,
 dkk: nil, eur: nil, gbp: nil, huf: nil, ils: nil, inr: nil, mxn: nil, nok: nil,
 nzd: nil, pln: nil, sek: nil, sgd: nil, usd: 100, zar: nil}
iex(2)> m.usd
100
```

**Review by csmb — APPROVED** [review]
Looks good to me!

**Pull request merged**